vexim/userchangesubmit.php

```diff
@@ -32,21 +32,23 @@
   }
 
 # Update the password, if the password was given
-  if (validate_password($_POST['clear'], $_POST['vclear'])) {
-    $cryptedpassword = crypt_password($_POST['clear']);
-    $query = "UPDATE users SET crypt=:crypt WHERE user_id=:user_id";
-    $sth = $dbh->prepare($query);
-    $success = $sth->execute(array(':crypt'=>$cryptedpassword, ':user_id'=>$_SESSION['user_id']));
-    if ($success) {
-      $_SESSION['crypt'] = $cryptedpassword;
-      header ("Location: userchange.php?userupdated");
-      die;
-    } else {
+  if (isset($_POST['clear']) && $_POST['clear']!=='') {
+    if (validate_password($_POST['clear'], $_POST['vclear'])) {
+      $cryptedpassword = crypt_password($_POST['clear']);
+      $query = "UPDATE users SET crypt=:crypt WHERE user_id=:user_id";
+      $sth = $dbh->prepare($query);
+      $success = $sth->execute(array(':crypt'=>$cryptedpassword, ':user_id'=>$_SESSION['user_id']));
+      if ($success) {
+        $_SESSION['crypt'] = $cryptedpassword;
+        header ("Location: userchange.php?userupdated");
+        die;
+      } else {
```

Comment:
Yeah. I guess we should either use `failupdated=$email_address`, or create a new error message for sql errors.
Or you can just delete these three lines, and let it fail as it does now. :) I still think we have a big rewrite pending anyway.